### DISCUSSION TITLE
Image component: preload with media attribute

### DISCUSSION THREAD

**Discussion by rmunch** [vercel/next.js#25171] [Ideas]
### Describe the feature you'd like to request

I've got a scenario where I display images only for a particular viewport. This is done by using CSS and media queries to hide and show the widget. I'd like to use the new Image component for these images to take advantage of optimizations, though I notice that when I mark these images as `priority`, they are preloaded for _all_ viewports (when they shouldn't be loaded at all). This impacts performance for viewports not showing the images.

### Describe the solution you'd like

I see that the link + preload tag supports a [media attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Preloading_content#including_media) that'd make it possible to limit the preload to a desired viewport, but it's not part of the Image component API. The first idea that comes to mind is to expose a new property to the Image component like `preloadMedia` or similar.

### Describe alternatives you've considered

I haven't dug into any workarounds at this point.

**Comment by marcelobaez** [comment]
I can't get the priority option to work. It renders a normal  tag instead of . How do you configured it? Are you using a custom loader?
